std/testing/asserts.ts

```diff
@@ -429,6 +429,44 @@ export function assertNotMatch(
   }
 }
 
+/**
+ * Make an assertion that `actual` object is a subset of `expected` object, deeply.
+ * If not, then throw.
+ */
+export function assertObjectMatch(
+  actual: Record<string | symbol, unknown>,
+  expected: Record<string | symbol, unknown>,
+): void {
+  type loose = Record<string | symbol, unknown>;
+  const seen = new WeakMap();
+  return assertEquals(
+    (function filter(a: loose, b: loose): loose {
+      // Prevent infinite loop with circular references with same filter
+      if ((seen.has(a)) && (seen.get(a) === b)) {
+        return a;
+      }
+      seen.set(a, b);
+      // Iterate through keys present in both actual and expected
+      // and filter recursively on object references
+      const filtered = {} as loose;
+      for (
+        const [key, value] of Object.entries(a).filter(([key]) => key in b)
```

Comment:
I don't think `Object.entries()` includes symbols. This would be shown if the tests included an erroneous case for symbols. You can use `[...Object.getOwnPropertyNames(a), ...Object.getOwnPropertySymbols(a)]` to get a complete list of keys.

Reply:
I replaced `Object.entries()` with your suggestion and it works like a charm, and changed `string|symbol` by `PropertyKey` which seems more accurate.

I got some trouble with using `symbol` as index signature so I casted them into `string` but not sure it's the cleanest way...

I wanted to add an erroneous case like you suggested, but seems that `assertEquals` does not handle `symbol`s perfectly currently (unless I missed something) : 
```ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts"
const foo = Symbol("foo")
Deno.test("symbol1", () => assertEquals({[foo]:"bar"}, {[foo]:"bar"}))
Deno.test("symbol2", () => assertEquals({[foo]:"bar"}, {[Symbol("foo")]:"bar"}))
```
```
running 2 tests
test symbol1 ... ok (2ms)
test symbol2 ... ok (1ms)

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (4ms)
```